### PR TITLE
Frontend Translation mixed up when switching languages

### LINKED ISSUE #16412: Frontend Translation mixed up when switching languages
This bug is triggered in various ways i will describe a easily reproducible way here:

- Start with a fresh installation of Prestashop 1.7.6.1
- Add 2 or more additional language packs (German and French in my case)
- Create a customer account
- Change customer firstname to a **invalid** one like "John.P" by direct SQL update
- Log in with the Customer on the Frontend
- Switch between the frontend languages

Now notice how some parts of the page will correctly update languages while others will remain in the previous language.

### PR DESCRIPTION
**Pull request by 23b**
Please review, this has not been tested in depth.

| Questions     | Answers
| ------------- | -------------------------------------------------------
| Branch?       | develop
| Description?  | In some cases smarty translation will show previous language setting.
| Type?         | bug fix
| Category?     | FO
| BC breaks?    | no
| Deprecations? | no
| Fixed ticket? | Fixes #16412
| How to test?  | See #16412 for reproduction.

---
This change is [Reviewable](https://reviewable.io/reviews/prestashop/prestashop/16439)

### CODE REVIEW

File: classes/Context.php [hunk 1]
```diff
@@ -378,7 +378,9 @@ public function updateCustomer(Customer $customer)
     public function getTranslator($isInstaller = false)
     {
         if (null !== $this->translator) {
-            return $this->translator;
+            if ($this->language->locale === $this->translator->getLocale()) {
```
Comment (PierreRambaud) [inline classes/Context.php]: Could be done in a single `if`

**Comment by colinegin** [comment]
Hello @23b, there has been one comment about your pull request, would you be able to fix it before we review it again ? 

Thanks !

**Review by Progi1984 — APPROVED** [review]

**Review by atomiix — APPROVED** [review]

**Review by PierreRambaud — APPROVED** [review]

**Comment by khouloudbelguith** [comment]
Hi @23b,

It is ok :heavy_check_mark: 
https://drive.google.com/file/d/1r28EpS26gmS0iWFKHJk7orzPcIiytRN0/view?usp=sharing

Thanks!

**Comment by Progi1984** [comment]
Thanks @23b & @khouloudbelguith

**Pull request merged**